Gems/ROS2/Code/Source/RobotImporter/URDF/URDFPrefabMaker.cpp

```diff
@@ -0,0 +1,332 @@
+        if (!createEntityResult.IsSuccess())
+        {
+            return createEntityResult;
+        }
+        AZ::EntityId entityId = createEntityResult.GetValue();
+        AZ::Entity* entity = AzToolsFramework::GetEntityById(entityId);
+
+        // Add ROS2FrameComponent - TODO: only for top level and joints
+        // TODO - add unique namespace to the robot's top level frame
+        const auto frameCompontentId = Utils::CreateComponent(entityId, ROS2FrameComponent::TYPEINFO_Uuid());
+        if (frameCompontentId)
+        {
+            auto* component = Utils::GetGameOrEditorComponent<ROS2FrameComponent>(entity);
+            AZ_Assert(component, "ROS2 Frame Component does not exist for %s", entityId.ToString().c_str());
+            component->SetFrameID(AZStd::string(link->name.c_str(), link->name.size()));
+        }
+        m_visualsMaker.AddVisuals(link, entityId);
+        m_collidersMaker.AddColliders(link, entityId);
+        m_inertialsMaker.AddInertial(link->inertial, entityId);
+        return AZ::Success(entityId);
+    }
+
+    void URDFPrefabMaker::AddRobotControl(AZ::EntityId rootEntityId)
+    {
+        const auto componentId = Utils::CreateComponent(rootEntityId, ROS2RobotControlComponent::TYPEINFO_Uuid());
+        if (componentId)
+        {
+            ControlConfiguration controlConfiguration;
+            TopicConfiguration subscriberConfiguration;
+            subscriberConfiguration.m_topic = "cmd_vel";
+            AZ::Entity* rootEntity = AzToolsFramework::GetEntityById(rootEntityId);
+            auto* component = Utils::GetGameOrEditorComponent<ROS2RobotControlComponent>(rootEntity);
+            component->SetControlConfiguration(controlConfiguration);
+            component->SetSubscriberConfiguration(subscriberConfiguration);
+        }
+    }
+
+    const AZStd::string& URDFPrefabMaker::GetPrefabPath() const
+    {
+        return m_prefabPath;
+    }
+
+    void URDFPrefabMaker::MoveEntityToDefaultSpawnPoint(const AZ::EntityId& rootEntityId)
+    {
+        auto spawner = ROS2::SpawnerInterface::Get();
+
+        if (spawner == nullptr)
+        {
+            AZ_TracePrintf("URDF Importer", "Spawner not found - creating entity in (0,0,0)") return;
+        }
+
+        auto entity_ = AzToolsFramework::GetEntityById(rootEntityId);
+        auto* transformInterface_ = entity_->FindComponent<AzToolsFramework::Components::TransformComponent>();
+
+        if (transformInterface_ == nullptr)
+        {
+            AZ_TracePrintf("URDF Importer", "TransformComponent not found in created entity") return;
+        }
+
+        auto pose = spawner->GetDefaultSpawnPose();
+
+        transformInterface_->SetWorldTM(pose);
+    }
+
+    AZStd::string URDFPrefabMaker::getStatus()
+    {
+        AZStd::string str;
+        AZStd::lock_guard<AZStd::mutex> lck(m_statusLock);
+        for (const auto& [entry, entry_status] : m_status)
```

Comment:
The `entry_status` variable should be lowerCamelCase
```suggestion
        for (const auto& [entry, entryStatus] : m_status)
```